source/subtree-query.cpp

```diff
@@ -0,0 +1,40 @@
+struct SubtreeSegment {
+	SegmentTree seg{{}};
+	vi sz, ord;
+	int curr_ord = 0;
+	int dfs(const vvi &ad, int p) {
+		ord[p] = curr_ord++;
+		sz[p] = 1;
+		for(int x:ad[p])
+			sz[p] += dfs(ad, x);

```

Comment:
Isn't this an infinite loop if the adjacency list has both directions (pretty common case)? We should probably keep a parent as `dfs` argument